File: apps/api/models/analytics.py
from pydantic import BaseModel


class AnalyticsResponse(BaseModel):
    productivity_score: float
    velocity_score: float
    bus_factor: float
    risk_score: float
    health_score: float


class AnalyticsData(BaseModel):
    productivity_score: float
    velocity_score: float
    bus_factor: float
    risk_score: float
    health_score: float

File: apps/dashboard/src/types/analytics.ts
export interface AnalyticsData {
  productivity_score: number;

  velocity_score: number;

  bus_factor: number;

  risk_score: number;

  health_score: number;
}

export interface Snapshot {
  id: number;

  repository_id: number;

  productivity_score: number;

  velocity_score: number;

  bus_factor: number;

  risk_score: number;

  health_score: number;

  recorded_at: string;
}

export interface InsightData {
  summary: string;

  recommendations: string[];

  risks: string[];
}